src/openrct2-ui/windows/Changelog.cpp

```diff
@@ -73,7 +73,11 @@ class ChangelogWindow final : public Window
         {
             throw std::runtime_error("Unable to open " + path);
         }
-        return std::string((std::istreambuf_iterator<char>(fs)), std::istreambuf_iterator<char>());
+        auto length = fs.tellg();
```

Comment:
Just realized, this will be probably be 0, you probably wanted something like this:
```
    fs.seekg (0, fs.end);
    auto length = fs.tellg();
    fs.seekg (0, fs.beg);
```

Reply:
Nice catch again. Given I have access to `beg` and `end`, I can just diff the two values.

Reply:
You might want to reconsider this
```
    static constexpr _Seekdir beg = _Seekbeg;
    static constexpr _Seekdir cur = _Seekcur;
    static constexpr _Seekdir end = _Seekend;
```
They are not offsets.

Reply:
I swear I know those things…